gcalcli/gcalcli.py

```diff
@@ -975,49 +978,53 @@ def _edit_event(self, event):
                 sys.exit(0)
 
             elif val.lower() == 't':
-                self.printer.msg('Title: ', 'magenta')
-                val = input()
+                val = get_input(self.printer, 'Title: ', STR_NOT_EMPTY)
                 if val.strip():
                     event['summary'] = val.strip()
 
             elif val.lower() == 'l':
-                self.printer.msg('Location: ', 'magenta')
-                val = input()
+                val = get_input(self.printer, 'Location: ', STR_ALLOW_EMPTY)
                 if val.strip():
                     event['location'] = val.strip()
 
             elif val.lower() == 'w':
-                self.printer.msg('When: ', 'magenta')
-                val = input().strip()
+                val = get_input(self.printer, 'When: ', PARSABLE_DATE).strip()
                 if val:
                     td = (event['e'] - event['s'])
                     length = ((td.days * 1440) + (td.seconds / 60))
+                    try:
+                        all_day = self.options.get('allday')
+                    except KeyError:
+                        all_day = None
                     try:
                         newStart, newEnd = utils.get_times_from_duration(
-                                val, length, self.options['allday'])
+                                val, length, all_day)
```

Comment:
Would you mind changing newStart and newEnd to new_start, new_end (and also on line 1005) while you're touching this line?